### PR TITLE
Fix RBM initialization

### PR DESCRIPTION
**Pull request by josephbowles**
Realized there was an error in the design of the RestrictedBoltzmannMachine. The class was calling fit from the inherited sklearn RBM, but this method does not call initialize so the attribute self.dim was not set. This leads to problems in hyperparameter opt that involve scores that need sampling (like MMD), since the pipeline will not call initialize.

The fix overwrites the fit method and calls initialize.

### CODE REVIEW

File: src/qml_benchmarks/models/energy_based_model.py [hunk 1]
```diff
@@ -90,11 +90,14 @@ def __init__(
         self.score_fn = score_fn
         self.mmd_kwargs = mmd_kwargs
 
-    def initialize(self, x: any = None):
-        self.fit(x[:1, ...])
```
Comment (josephbowles) [inline src/qml_benchmarks/models/energy_based_model.py:94]: I removed this because I find it a bit messy; and it could take a long time in principle. A better way would be to randomly initialize the parameters of the sklearn RBM without calling fit.

File: src/qml_benchmarks/models/energy_based_model.py [hunk 2]
```diff
@@ -128,4 +131,8 @@ def score(self, X: np.ndarray, y: np.ndarray) -> float:
         if self.score_fn == 'pseudolikelihood':
             return float(np.mean(super().score_samples(X)))
         elif self.score_fn == 'mmd':
-            return float(-mmd_loss(X, self.sample(self.mmd_kwargs['n_samples']), self.mmd_kwargs['sigma']))
\ No newline at end of file
+            sigma = self.mmd_kwargs['sigma']
+            sigmas = [sigma] if isinstance(sigma, (int, float)) else sigma
+            score = np.mean([mmd_loss(X, self.sample(self.mmd_kwargs['n_samples'],
+                                                     self.mmd_kwargs['n_steps']), sigma) for sigma in sigmas])
+            return float(-score)
```
Comment (josephbowles) [inline src/qml_benchmarks/models/energy_based_model.py:138]: If one gives a list of sigmas, the mean of the mmd values is returned as the score now. Behaviour is the same as before otherwise.
